client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -105,10 +101,20 @@ public void deleteReaderGroup(String groupName) {
         StateSynchronizer<ReaderGroupState> synchronizer = clientFactory.createStateSynchronizer(NameUtils.getStreamForReaderGroup(groupName),
                 new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         synchronizer.fetchUpdates();
-        UUID groupId = synchronizer.getState().getConfig().getReaderGroupId();
-        long generation = synchronizer.getState().getConfig().getGeneration();
-        getAndHandleExceptions(controller.deleteReaderGroup(scope, groupName, groupId, generation),
-                RuntimeException::new);
+        ReaderGroupConfig syncConfig = synchronizer.getState().getConfig();
+        if (ReaderGroupConfig.DEFAULT_UUID.equals(syncConfig.getReaderGroupId())
+            && ReaderGroupConfig.DEFAULT_GENERATION == syncConfig.getGeneration()) {
+            // migrate RG case
+            getAndHandleExceptions(controller.createReaderGroup(scope, groupName,
```

Comment:
Yes. I thought of just deleting the State Synchronizer Stream inside this if block, but that won't handle the case where the ReaderGroup is created on Controller, but update to StateSynchronizer has failed and so SS has readerGroupId as default.
Hence added this createRG() call to be sure we don't leave the RG on Controller by only deleting state-synchronizer stream for the RG.

Reply:
so shouldnt we just call delete on controller and if that fails with NotFound like exeption (which it should not assuming delete is idempotent) then we handle and ignore it and proceed with deleting in state sync.. i dnt see why we should add it to controller and then delete it.

Reply:
deleteReaderGroup() API needs `ReaderGroupId` and `generation` of the ReaderGroup.
Added a getReaderGroupConfig() call instead of createRG() and if ReaderGroup is not found, it just deletes the StateSynchronizer Stream.